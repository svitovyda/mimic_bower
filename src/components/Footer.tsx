import * as React from "react";

export const Footer: React.FC = () => {
  return <div></div>;
};
