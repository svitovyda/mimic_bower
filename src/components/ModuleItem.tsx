import type { Package } from "../models/Package";
import * as React from "react";

export interface ModuleItemProp {
  packageElement: Package;
}

export const ModuleItem: React.FC<ModuleItemProp> = ({ packageElement }: ModuleItemProp) => {
  return (
    <div>
      {packageElement.name} {packageElement.owner} {packageElement.stars}
    </div>
  );
};
