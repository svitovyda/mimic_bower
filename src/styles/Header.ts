import LogoSvg from "./../assets/logo.svg";
import styled from "styled-components";

export const HeaderContainer = styled.div`
  display: block;
  position: relative;
  max-width: 1000px;
  margin: 0 auto;
  padding: 0 10px;
  font-weight: bold;
`;

export const HeaderTitle = styled.h1`
  font-size: 64px;
  letter-spacing: -0.025em;
  margin: 0;
  padding: 0;
  padding-top: 26px;
  padding-left: 200px;

  @media only screen and (max-width: 680px) {
    display: inline-block;
    font-size: 32px;
    padding: 7px 0 5px 0;
  }
`;

export const HeaderSubtitle = styled.h4`
  margin: -20px 0 0 203px;
  font-size: 19px;

  @media only screen and (max-width: 680px) {
    display: none;
  }
`;

export const HeaderLogo = styled(LogoSvg)`
  display: block;
  position: absolute;
  left: 30px;
  top: 15px;
  transition: all 0.2s ease;

  :hover {
    transform: scale(1.1);
  }

  @media only screen and (max-width: 680px) {
    position: relative;
    display: inline;
    width: 42px;
    height: 42px;
    left: auto;
    top: 7px;
  }
`;
