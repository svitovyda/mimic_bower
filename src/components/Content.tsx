import { useContentFetcher } from "../hooks/contentFetcher";
import { ModuleItem } from "./ModuleItem";
import { LoaderAnimation } from "./ui/LoaderAnimation";
import * as React from "react";

export const Content: React.FC = () => {
  const [state, setQuery] = useContentFetcher("");

  React.useEffect(() => {
    setQuery("jquery");
  }, [setQuery]);

  if (state.isLoading) return <LoaderAnimation />;
  if (state.isError) return <div>Some error occurred, try again later!</div>;
  return (
    <div>
      {state.data.map((p) => (
        <ModuleItem key={p.name} packageElement={p} />
      ))}
    </div>
  );
};
