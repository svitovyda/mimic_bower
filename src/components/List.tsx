import type { Package } from '../models/Package';
import { ModuleItem } from './ModuleItem';
import * as React from 'react';

export interface ListProps {
  data: Package[];
}

export const List: React.FC<ListProps> = ({ data }: ListProps) => {
  return (
    <div>
      {data.map((p, i) => (
        <ModuleItem key={`${i}-${p.name}-${p.platform}`} packageElement={p} />
      ))}
    </div>
  );
};
