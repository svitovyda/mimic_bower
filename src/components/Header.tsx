import { HeaderContainer, HeaderLink, HeaderLogo, HeaderSubtitle, HeaderTitle } from "../styles/Header";
import * as React from "react";

export const Header: React.FC = () => {
  return (
    <header>
      <HeaderContainer>
        <HeaderLogo />
        <HeaderTitle>Bower Search</HeaderTitle>
        <HeaderSubtitle>
          Powered by{" "}
          <HeaderLink href="https://libraries.io/" target="_blank" rel="noreferrer">
            libraries.io
          </HeaderLink>
        </HeaderSubtitle>
      </HeaderContainer>
    </header>
  );
};
