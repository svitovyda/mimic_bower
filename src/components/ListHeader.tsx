/* eslint-disable jsx-a11y/anchor-is-valid */
import type { SortType } from '../services/ApiServise';
import { ListHeaderContainer, NameLink, OwnerLink, StarsLink } from '../styles/ListHeader';
import * as React from 'react';

export interface ListHeaderProps {
  onSort: (sortBy?: SortType) => void;
}

export const ListHeader: React.FC<ListHeaderProps> = React.memo(({ onSort }: ListHeaderProps) => {
  return (
    <ListHeaderContainer>
      <NameLink onClick={() => onSort()}>Name</NameLink>
      <OwnerLink>Owner</OwnerLink>
      <StarsLink onClick={() => onSort('stars')}>Stars</StarsLink>
    </ListHeaderContainer>
  );
});

ListHeader.displayName = 'ListHeader';
