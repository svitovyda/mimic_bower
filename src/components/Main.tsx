import { Content } from "./Content";
import { Search } from "./Search";
import * as React from "react";

export const Main: React.FC = () => {
  const [query, setQuery] = React.useState<string>("");

  const onNewQuery = React.useCallback(
    (newQuery: string) => {
      setQuery(newQuery);
    },
    [setQuery]
  );

  return (
    <div>
      <Search onQueryChanged={onNewQuery} />
      <Content query={query} />
    </div>
  );
};
