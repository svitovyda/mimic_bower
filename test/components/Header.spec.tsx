import { Header } from "../../src/components/Header";
import { render } from "@testing-library/react";
import "jest-styled-components";
import * as React from "react";

describe("Header", () => {
  it("renders without error", () => {
    const component = render(<Header />);

    expect(component).toMatchSnapshot();
  });
});
