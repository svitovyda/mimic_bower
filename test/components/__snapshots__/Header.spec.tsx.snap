// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders without error 1`] = `
{
  "asFragment": [Function],
  "baseElement": .c0 {
  display: block;
  position: relative;
  max-width: 1000px;
  margin: 0 auto;
  padding: 0 10px;
  font-weight: bold;
}

.c2 {
  font-size: 64px;
  -webkit-letter-spacing: -0.025em;
  -moz-letter-spacing: -0.025em;
  -ms-letter-spacing: -0.025em;
  letter-spacing: -0.025em;
  margin: 0;
  padding: 0;
  padding-top: 26px;
  padding-left: 200px;
}

.c3 {
  margin: -20px 0 0 203px;
  font-size: 19px;
}

.c4 {
  color: #008ec4;
  outline-style: none;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1 {
  display: block;
  position: absolute;
  left: 30px;
  top: 15px;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
}

.c1:hover {
  -webkit-transform: scale(1.1);
  -ms-transform: scale(1.1);
  transform: scale(1.1);
}

@media only screen and (max-width:680px) {
  .c2 {
    display: inline-block;
    font-size: 32px;
    padding: 7px 0 5px 0;
  }
}

@media only screen and (max-width:680px) {
  .c3 {
    display: none;
  }
}

@media only screen and (max-width:680px) {
  .c1 {
    position: relative;
    display: inline;
    width: 42px;
    height: 42px;
    left: auto;
    top: 7px;
  }
}

<body>
    <div>
      <header>
        <div
          class="c0"
        >
          <mock
            class="c1"
          />
          <h1
            class="c2"
          >
            Bower Search
          </h1>
          <h4
            class="c3"
          >
            Powered by
             
            <a
              class="c4"
              href="https://libraries.io/"
              rel="noreferrer"
              target="_blank"
            >
              libraries.io
            </a>
          </h4>
        </div>
      </header>
    </div>
  </body>,
  "container": .c0 {
  display: block;
  position: relative;
  max-width: 1000px;
  margin: 0 auto;
  padding: 0 10px;
  font-weight: bold;
}

.c2 {
  font-size: 64px;
  -webkit-letter-spacing: -0.025em;
  -moz-letter-spacing: -0.025em;
  -ms-letter-spacing: -0.025em;
  letter-spacing: -0.025em;
  margin: 0;
  padding: 0;
  padding-top: 26px;
  padding-left: 200px;
}

.c3 {
  margin: -20px 0 0 203px;
  font-size: 19px;
}

.c4 {
  color: #008ec4;
  outline-style: none;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1 {
  display: block;
  position: absolute;
  left: 30px;
  top: 15px;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
}

.c1:hover {
  -webkit-transform: scale(1.1);
  -ms-transform: scale(1.1);
  transform: scale(1.1);
}

@media only screen and (max-width:680px) {
  .c2 {
    display: inline-block;
    font-size: 32px;
    padding: 7px 0 5px 0;
  }
}

@media only screen and (max-width:680px) {
  .c3 {
    display: none;
  }
}

@media only screen and (max-width:680px) {
  .c1 {
    position: relative;
    display: inline;
    width: 42px;
    height: 42px;
    left: auto;
    top: 7px;
  }
}

<div>
    <header>
      <div
        class="c0"
      >
        <mock
          class="c1"
        />
        <h1
          class="c2"
        >
          Bower Search
        </h1>
        <h4
          class="c3"
        >
          Powered by
           
          <a
            class="c4"
            href="https://libraries.io/"
            rel="noreferrer"
            target="_blank"
          >
            libraries.io
          </a>
        </h4>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
